plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/AbstractXmlValidatorTest.java

```diff
@@ -29,58 +33,99 @@
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.Path;
+import org.eclipse.jst.common.project.facet.core.JavaFacet;
+import org.eclipse.jst.j2ee.web.project.facet.WebFacetUtils;
+import org.eclipse.wst.common.project.facet.core.IProjectFacetVersion;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
+import org.eclipse.wst.validation.ValidationFramework;
+import org.eclipse.wst.validation.Validator;
 import org.junit.After;
-import org.junit.BeforeClass;
+import org.junit.Before;
 import org.junit.ClassRule;
 import org.junit.Test;
 import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
 
 public class AbstractXmlValidatorTest {
 
-  private final String ELEMENT_MESSAGE = "Project ID should be specified at deploy time.";
-  private static IResource resource;
-  private static IProject project;
+  private static final IProjectFacetVersion APPENGINE_STANDARD_FACET_VERSION_1 =
+      ProjectFacetsManager.getProjectFacet(AppEngineStandardFacet.ID).getVersion("1");;
+  private IFile resource;
 
-  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator();
+  @ClassRule public static TestProjectCreator appEngineStandardProjectCreator =
+      new TestProjectCreator().withFacetVersions(Lists.newArrayList(JavaFacet.VERSION_1_7,
+          WebFacetUtils.WEB_25, APPENGINE_STANDARD_FACET_VERSION_1));
+  
+  @ClassRule public static TestProjectCreator dynamicWebProjectCreator =
+      new TestProjectCreator().withFacetVersions(Lists.newArrayList(JavaFacet.VERSION_1_7,
+                                                                    WebFacetUtils.WEB_25));
 
-  @BeforeClass
-  public static void setUp() throws CoreException {
-    project = projectCreator.getProject();
+  @Before
+  public void setUp() throws CoreException {
+    IProject project = dynamicWebProjectCreator.getProject();
     createFolders(project, new Path("src/main/webapp/WEB-INF"));
```

Comment:
AFAIK, one needs to specify a config to create the DD (`web.xml`) when adding the Dynamic Web facet. The following returns true, true, and false:

```java
    System.out.println("WebContent: " + project.getFolder("WebContent").exists());
    System.out.println("WebContent/WEB-INF: " + project.getFolder("WebContent/WEB-INF").exists());
    System.out.println("WebContent/WEB-INF/web.xml: " + project.getFolder("WebContent/WEB-INF/web.xml").exists());
```